File: samples/snippets/visualbasic/VS_Snippets_VBCSharp/VbPowerPacksLineHitTest/VB/LineHitTestForm.vb
Public Class LineHitTestForm

    ' <Snippet1>
    Private Sub LineHitTestForm_PreviewKeyDown(
        ByVal sender As Object, 
        ByVal e As System.Windows.Forms.PreviewKeyDownEventArgs
      ) Handles Me.PreviewKeyDown

        If e.KeyCode = Keys.Space Then
            Dim px As Integer
            Dim py As Integer
            Dim hit As Boolean
            px = LineHitTestForm.MousePosition.X
            py = LineHitTestForm.MousePosition.Y
            hit = LineShape1.HitTest(px, py)
            MsgBox(CStr(hit))
        End If
    End Sub
    ' </Snippet1>
End Class

File: samples/snippets/csharp/VS_Snippets_VBCSharp/VbPowerPacksLineHitTest/CS/LineHitTestForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace VbPowerPacksLineHitTestCS
{
    public partial class LineHitTestForm : Form
    {
        public LineHitTestForm()
        {
            InitializeComponent();
        }

        // <Snippet1>
        private void LineHitTestForm_PreviewKeyDown(object sender, 
            System.Windows.Forms.PreviewKeyDownEventArgs e)
        {
            if (e.KeyCode == Keys.Space)
            {
                int px;
                int py;
                bool hit;
                string result;
                px = LineHitTestForm.MousePosition.X;
                py = LineHitTestForm.MousePosition.Y;
                hit = lineShape1.HitTest(px, py);
                result = hit.ToString();
                MessageBox.Show(result);
            }
        }
        // </Snippet1>
    }
}
